File: singly_linked_lists/backremoveadd.js
class Node{
    constructor(value){
        this.head = value
        this.next = null
    }
}

class SLL{
    constructor(){
        this.head = null
    }
    back(){
        let result = 0
        let node = this.head
        while(node !== null){
            result = node.data
            node = node.next
        }
        return result
    }
    removeBack(){
        let node = this.head
        while(node.next.next !== null){
            node = node.next
        }
        node.next = null
    }
    addBack(value){
        let newNode = new Node(value)
        let node = this.head
        while(node.next != null){
            node = node.next
        }
        node.next = newNode
        return
    }
}

File: singly_linked_lists/python/backremoveadd.py
class Node:
    def __init__(self, value=None):
        self.value = value
        self.next = None

class SLL:
    def __init__(self):
        self.head = None
        self.tail = None

    def addFront(self, value):
        newNode = Node(value)
        if self.head is None:
            self.head = newNode
            self.tail = newNode
        else:
            newNode.next = self.head
            self.head = newNode

    def display(self):
        runner = self.head
        result = ""
        while runner != None:
            if runner.next != None:
                result = result + str(runner.value) + ", "
            else:
                result = result + str(runner.value)
            runner = runner.next
        return result

    def back(self):
        runner = self.head
        while runner != None:
            if runner.next == None:
                return runner.value
            runner = runner.next

    def removeBack(self):
        runner = self.head
        while runner != None:
            if runner.next.next == None:
                runner.next = None
                self.tail = runner
            runner = runner.next
        return self.display()

    def addBack(self, value):
        newNode = Node(value)
        # runner = self.head
        # while runner != None:
        #     if runner.next == None:
        #         runner.next = newNode
        #         return runner.value
        #     runner = runner.next
        newNode.next = None
        self.tail.next = newNode
        self.tail = newNode

SLL1 = SLL()
SLL1.addFront(5)
SLL1.addFront(4)
SLL1.addFront(3)
SLL1.addFront(2)
SLL1.addFront(1)
SLL1.addFront(0)
SLL1.addBack(6)
SLL1.addBack(8)
SLL1.addBack(7)
SLL1.removeBack()
print(SLL1.display())